modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -365,7 +371,11 @@ protected HttpMessage createMessage(String[] initialLine) throws Exception {
                     );
             }
             // combines the HTTP message pieces into a single full HTTP request (with headers and body)
-            final HttpObjectAggregator aggregator = new Netty4HttpAggregator(handlingSettings.maxContentLength());
+            final HttpObjectAggregator aggregator = new Netty4HttpAggregator(
+                handlingSettings.maxContentLength(),
+                httpPreRequest -> enabled.get() == false
+                    || (httpPreRequest.uri().contains("_bulk") == false || httpPreRequest.uri().contains("_bulk_update"))
```

Comment:
Can we make this:

```suggestion
                    || (httpPreRequest.uri().endsWith("/_bulk") == false && httpPreRequest.uri().contains("/_bulk?") == false)
```

instead? Seems slightly safer than the anti test for `_bulk_update`

Reply:
I _think_ the uri in this scenario contains query params. But I will look into this. This particular piece of work is quite unfortunate and hopefully we can find a better way either resolving the handler (doing full path analysis) earlier or if we change how we do all aggregation.

Reply:
There is even another route that I am having to handle now (old: `/_xpack/monitoring/_bulk`).

Reply:
https://elasticco.atlassian.net/browse/ES-9309